From 38b5ce6e357e5c55fa0aa01ba748bf6672c97b94 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Fri, 12 Jul 2013 07:38:39 +0530
Subject: Patch Fixes compilation issues - Enabling offload in PDK

BZ: 123052

This patch fixes the compilation issues while porting offload in MR2

Change-Id: Iec108ca8ae6127fd9555666177ccfeacb4dda10f
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 audio/AudioPolicyManagerBase.cpp                 |    3 ++-
 include/hardware_legacy/AudioPolicyManagerBase.h |    2 --
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 56c745a..1f1ebc3 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -1137,7 +1137,8 @@ void AudioPolicyManagerBase::releaseOutput(audio_io_handle_t output)
             mpClientInterface->setParameters(0, param.toString(), 0);
 #endif
         // Close offload output only if ref count is zero.
-        if (outputDesc->refCount() == 0) {
+        int refCount = outputDesc->mRefCount[index];
+        if (refCount == 0) {
             ALOGV("releaseOutput: closing output");
             mpClientInterface->closeOutput(output);
             delete mOutputs.valueAt(index);
diff --git a/include/hardware_legacy/AudioPolicyManagerBase.h b/include/hardware_legacy/AudioPolicyManagerBase.h
index a87f95a..f6aedf9 100644
--- a/include/hardware_legacy/AudioPolicyManagerBase.h
+++ b/include/hardware_legacy/AudioPolicyManagerBase.h
@@ -606,11 +606,9 @@ protected:
         uint32_t        mTestLatencyMs;
 #endif //AUDIO_POLICY_TEST
 
-#ifdef BGM_ENABLED
         /*flag to keep track of background music*/
         bool     mIsBGMEnabled;
         audio_io_handle_t mBGMOutput;
-#endif /*BGM_ENABLED*/
 
 private:
         static float volIndexToAmpl(audio_devices_t device, const StreamDescriptor& streamDesc,
-- 
1.7.4.1

